docs/01-app/02-guides/mcp.mdx

```diff
@@ -0,0 +1,132 @@
+
+## Enabling the MCP server
+
+To enable the experimental MCP server in your Next.js application, use Next.js v16 or above. Then add the following to your `next.config.js`:
+
+```js filename="next.config.js"
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    mcpServer: true,
+  },
+}
+
+module.exports = nextConfig
+```
+
+### Configuring Next DevTools MCP
+
+When you enable the MCP server, configure [`next-devtools-mcp`](https://www.npmjs.com/package/next-devtools-mcp) in a `.mcp.json` file at the root of your project to enhance the agent's capabilities with additional development tools:
+
+```json filename=".mcp.json"
+{
+  "mcpServers": {
+    "next-devtools": {
+      "command": "npx",
+      "args": ["-y", "next-devtools-mcp@latest"]
+    }
+  }
+}
+```
+
+See the [next-devtools-mcp documentation](https://www.npmjs.com/package/next-devtools-mcp) for more configuration options and features.
+
+## What the MCP server provides
+
+With `experimental.mcpServer` enabled and `next-devtools-mcp` configured, agents get access to:
+
+- **Next.js Knowledge Base** - Comprehensive Next.js documentation and best practices
+- **Debugging Insights** - Real-time error detection, build diagnostics, logs, and runtime information from your dev server
+- **Migration Tools** - Automated helpers for migrating your application, including upgrading to Next.js 16 with codemods and enabling Cache Components with automated setup and error detection
+
+## Using with agents
+
+When you have the MCP server enabled, MCP-compatible Coding Agents can automatically discover and connect to your Next.js application with the experimental `mcpServer` flag enabled. This provides agents with rich context about your project.
+
+### Benefits for agent-assisted development
+
+With the MCP server enabled, agents can:
+
+- **Understand your project layout** - Know exactly where routes, components, and utilities are located
+- **Make context-aware suggestions** - Recommend the right place to add new features based on your existing structure
+- **Query live application state** - Check current configuration, routes, and middleware during development
+- **Reduce context overhead** - No need to manually explain your app structure each conversation
+- **Provide accurate implementations** - Generate code that follows your project's patterns and conventions
+
+## Development workflow
+
+1. **Start your Next.js server** with the MCP server enabled:
+
+```bash filename="Terminal"
+npm run dev
+```
+
+2. **Connect your Coding Agent** to the running Next.js instance
+
+3. **Load your application in the browser** to trigger compilation and rendering. The MCP tools work by analyzing the compiled and rendered state of your application.
+
+4. **Query your agent** for insights and diagnostics:
+
+```
+What errors are currently in my application?
```

Comment:
is that the list of _all_ possible commands? we really need the docs to list eveyrthing. otherwise it's too vague.